### PR TITLE
Drive to ball to main

### PR DESCRIPTION
**Pull request by Spenceroni-Pepperoni**

### CODE REVIEW

**Review by KevinH45 — APPROVED** [review]

File: src/main/java/frc/robot/NetworkTables/NetworkTablesBase.java [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 
 //Class
-public class NetworkTablesBase {
+public class NetworkTablesBase extends Thread {
```
Comment (liam-a-wms) [inline src/main/java/frc/robot/NetworkTables/NetworkTablesBase.java:11]: To properly extend the `Thread` class, add the `@Override` annotation to the `run()` method and get it out of the constructor.

File: src/main/java/frc/robot/NetworkTables/NetworkTablesBase.java [hunk 2]
```diff
@@ -28,22 +28,11 @@ public void run() {
...
-    }
-  }
+    inst.startDSClient(); // recommended if running on DS computer; this gets the robot IP from the DS
+
+    double x = SmartDashboard.getNumber("front_ball_x", 0);
+    double y = SmartDashboard.getNumber("front_ball_y", 0);
 
-}
\ No newline at end of file
+    System.out.println("X: " + x + " Y: " + y);
```
Comment (liam-a-wms) [inline src/main/java/frc/robot/NetworkTables/NetworkTablesBase.java]: Is this absolutely necessary, or just flooding the console + log for no reason?

File: src/main/java/frc/robot/commands/CameraDrive.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Drivetrain;
+
+
+public class CameraDrive extends CommandBase {
```
Comment (liam-a-wms) [inline src/main/java/frc/robot/commands/CameraDrive.java]: This command is unused, and looks like a duplicate of `CameraDriveCommand`. If it's not going to be used then delete it.

File: src/main/java/frc/robot/commands/CameraDriveCommand.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Drivetrain;
+import java.lang.Math;
+
+public class CameraDriveCommand extends CommandBase {
+  /** Creates a new CameraDrive. */
+  Drivetrain drive;
+  Timer diveTimeout = new Timer();
+
+    double ballX;
```
Comment (liam-a-wms) [inline src/main/java/frc/robot/commands/CameraDriveCommand.java]: Use consistent spacing. VS Code has a built-in formatter you can use for this.

File: src/main/java/frc/robot/commands/CameraDriveCommand.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {}
+
+  // Called every time the scheduler runs while the command is scheduled.
+
+double XDeadzone = 6.5; // inches
+double YDeadzone = 4;
+double forwardDriveSpeed = 0.6;
```
Comment (liam-a-wms) [inline src/main/java/frc/robot/commands/CameraDriveCommand.java]: If this is a constant then it should be `static final` and in `ALL_CAPS`. Class-level variables and constants should also be at the top, below the class declaration, not under the constructor.

File: src/main/java/frc/robot/commands/CameraDriveCommand.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+            moveXValue = 4;
+            if (ballX < 0){
+              moveXValue=moveXValue*-1;
+            }else{
+              moveXValue = moveXValue * 1;
+            }
+          }else{
+            moveXValue = ballX;
+          }
+          drive.setMotors((0.05)*(moveXValue),(-0.05)*(moveXValue));
```
Comment (liam-a-wms) [inline src/main/java/frc/robot/commands/CameraDriveCommand.java]: Speed multiplier should be a constant to make it easier to adjust. It does not need to be in the main `Constants` file, so placing it below the class declaration should be fine.

**Review by liam-a-wms — CHANGES_REQUESTED** [review]

**Review by SixtyFiveOhTwo — APPROVED** [review]

**Pull request merged**